shared/src/native-src/filesystem.h

```diff
@@ -0,0 +1,11 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+
+#ifdef LINUX
+#include "filesystem.hpp"
+namespace fs = ghc::filesystem;
```

Comment:
> That seems risky to rely on #include <filesystem.h> rather than #include <filesystem>. 

do you mean in terms of header file resolution ?

> Can't we merge this filsystem.h into filesystem.hpp, have this #ifdef LINUX there, and even possibly rename filesystem.hpp to fs.hpp or something more custom?

We can, but I was under the impression that we should keep the `external` file as-is (here `filesystem.hpp`) and wrap around it (hence the `filesystem.h`)

Reply:
renaming `filesystem.h` into `dd_filesystem.h` would make more sense

Reply:
> can you be more specific please ? Not sure I understand the risk and what would be the move to face the ABI issue.

It's more a question of possible typo and how it would impact what files gets included. If you do `#include <filesystem>`, it's going to be the C++ style kind of include so it will IIRC probe for `filesystem.hpp` rather than `filesystem.h`. But if you do specify `#include <filesystem.h>` then it will include `filesystem.h` and not `filesystem.hpp`. So it's a small typo to make but a possible big impact as to what code gets included (the system one or the local one).

Overall, like you mention, it would even be better to have a `dd_filesystem.hpp` or `fs.hpp` (as long as it's not `filesystem.h*`), so that you need to use `#include "fs.hpp"` (note the `"..."` instead of`<...>`) to prioritize local include files over system include files.

Reply:
got it!